### DISCUSSION TITLE
Ways to manage the quality of contributions done by employees from an organization to open source communities

### DISCUSSION THREAD

**Discussion by anajsana** [todogroup/ospology#30] [OSPO Discussions]
This is an interesting question I got during the OSPO live Q&A at GitHub Universe. IMHO, there is not just one answer, so maybe more folks here can share their thoughts on this topic 😄

**Comment by bdemers** [comment]
For actual code quality, I don't think there is anything specific to OSS vs a private/corporate project.
I think much of the difference is around communication and expectations.

Community contributions are likely to take longer, they may require more discussion.

Don't view code review comments as negative, even if they seem this way. There are many reasons for this (opinion):
- Personal attachment to code
- Text only reviews are often short and to the point 
- Potential language differences between the reviewer and reviewee
- Your (or companies) timeline for getting the fix merged doesn't match the OSS projects
- No prior relationship with reviewers. It may be harder to take feedback from a stranger.


Other thoughts:
- Make sure you read contrib guides
- Find out if you need to sign a contributors agreement
- Communicate with the project, do they have mailing or discussion lists? 
- Be descriptive, pull requests and commits need to contain enough information for a stranger to understand the problem

Most of these are not specific to OSS projects, working on quality contributions inside your company will also help contributions outside.

**Comment by hyandell** [comment]
I was recently looking at someone's activity on a project as a part of a promotion review, and I think it scales to the company as a whole. The focus is also on repeat-contributions, not on a single contribution.

Some of the things I was looking at (many of these are overlapping concepts):

* Are they doing the janitor work as well as contributing new code.
* Are they contributing to the conversations about how to run the project, not just the code of the project.
* Do they have breadth of skills within the project (code, but also issue management or doc writing or licensing or build or ...). 
* Are they providing a unique skill set to the project. For example, they are the expert when it comes to running on ARM chips.
* How are they interacting with others on code reviews and conversations. Do they have both the backbone to make their point and the humility to accept others' points.
* Are the points they raise leading to change.
* Do they compromise. 
* Are they creating happiness. [running projects has a cost, project owners should not be happy to get good code that they had to crawl over broken glass for because someone is an ass]
* Do they exhibit desire to take on responsibility. 
* Do their contributions match the expectation of the project's guidelines - are they signing the DCO correctly, do they have the necessary testing, are they causing license confusion or not closing out their issues etc.
* Are they starting to exhibit leadership on an aspect of the project.
* Are they an ambassador for the project within the company.

This is all tempered with whether the project allows the space for a contributor to do those things. And I am sure there are many things I'm not thinking about :)
